test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -976,6 +977,385 @@ public static CommandRunnerResult RunRestore(SimpleTestPathContext pathContext,
+
+                var externalRepositoryPath = pathContext.PackageSource;
+                var contosoRepositoryPath = pathContext.UserPackagesFolder;
+
+                var configPath = Path.Combine(pathContext.WorkingDirectory, "nuget.config");
+                SettingsTestUtils.CreateConfigurationFile(configPath, $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""ExternalRepository"" value=""{externalRepositoryPath}"" />
+    <add key=""InternalRepository"" value=""{contosoRepositoryPath}"" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key=""ExternalRepository"">
+            <namespace id=""External.*"" />
+            <namespace id=""Others.*"" />
+        </packageSource>
+        <packageSource key=""InternalRepository"">
+            <namespace id=""Contoso.*"" />  <!--Contoso.A package doesn't exist Contoso repository, so restore should fail-->
+            <namespace id=""Test.*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+
+                var ExternalA = new SimpleTestPackageContext()
+                {
+                    Id = "Contoso.A",  // Initial version had package id conflict with Contoso repository
+                    Version = "1.0.0"
+                };
+                ExternalA.Files.Clear();
+                ExternalA.AddFile("lib/net461/externalA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalA);
+
+                var ExternalB = new SimpleTestPackageContext()
+                {
+                    Id = "External.B",  // name conflict resolved.
+                    Version = "2.0.0"
+                };
+                ExternalB.Files.Clear();
+                ExternalB.AddFile("lib/net461/externalB.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalB);
+
+                Util.CreateFile(Path.GetDirectoryName(projectA.ProjectPath), "packages.config",
+@"<packages>
+  <package id=""Contoso.A"" version=""1.0.0"" targetFramework=""net461"" />
+  <package id=""External.B"" version=""2.0.0"" targetFramework=""net461"" />
+</packages>");
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                // Act
+                var result = RunRestore(pathContext, _failureExitCode);
+
+                // Assert
+                Assert.Contains("Unable to find version '1.0.0' of package 'Contoso.A'", result.Errors);
+            }
+        }
+
+        [Fact]
+        public async Task Restore_PackageNamespacePartialPrefixes_Succeed()
```

Comment:
nit: Partial prefix is....interesting :) 

It's still a prefix. i don't find it too important that `NuGet.*` and `NuGet*` are both tested.

Reply:
Removed.